### PR TITLE
Make documentation and code improvements to PWA rules and parser

### PR DESCRIPTION
**Pull request by alrra**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messagess)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

### CODE REVIEW

File: packages/rule-manifest-is-valid/tests/tests.ts [hunk 1]
```diff
@@ -56,42 +54,39 @@ const defaultTests: Array<RuleTest> = [
         }
     },
     {
-        name: `Web app manifest is specified and the 'background_color' and 'theme_color'' properties are not valid`,
-        reports: [
-            { message: `'background_color' property value ('invalid') is invalid` },
-            { message: `'theme_color' property value ('invalid') is invalid` }
-        ],
+        name: `Web app manifest is specified and the 'background_color' and 'theme_color'' properties are are valid`,
```
Comment (molant) [inline packages/rule-manifest-is-valid/tests/tests.ts]: Remove extra `'` at the end of `'theme_color'` and extra `are`

File: packages/rule-manifest-is-valid/tests/tests.ts [hunk 2]
```diff
@@ -56,42 +54,39 @@ const defaultTests: Array<RuleTest> = [
...
-                theme_color: 'invalid'
+                background_color: 'red',
+                theme_color: '#ff0000'
                 /* eslint-enable camelcase */
             })
         }
     },
     {
-        name: `Web app manifest is specified and the 'background_color' and 'theme_color'' properties are not supported`,
+        name: `Web app manifest is specified and the 'background_color' and 'theme_color'' properties are not valid`,
```
Comment (molant) [inline packages/rule-manifest-is-valid/tests/tests.ts]: extra `'`

File: packages/rule-manifest-is-valid/tests/tests.ts [hunk 3]
```diff
@@ -56,42 +54,39 @@ const defaultTests: Array<RuleTest> = [
...
             })
         }
-    }
-];
-
-const testsForHexWithAlphaSupport: Array<RuleTest> = [
+    },
     {
-        name: `Web app manifest is specified and the 'background_color' and 'theme_color'' property values are supported because of the targeted browsers`,
+        name: `Web app manifest is specified and the 'background_color' and 'theme_color'' properties are not supported`,
```
Comment (molant) [inline packages/rule-manifest-is-valid/tests/tests.ts]: Extra `'`

File: packages/rule-manifest-is-valid/tests/tests.ts (hunk 1 above)
Comment (alrra) [reply packages/rule-manifest-is-valid/tests/tests.ts]: Done.

File: packages/rule-manifest-is-valid/tests/tests.ts (hunk 2 above)
Comment (alrra) [reply packages/rule-manifest-is-valid/tests/tests.ts]: Done.

File: packages/rule-manifest-is-valid/tests/tests.ts (hunk 3 above)
Comment (alrra) [reply packages/rule-manifest-is-valid/tests/tests.ts]: Done.

**Pull request closed without merging**